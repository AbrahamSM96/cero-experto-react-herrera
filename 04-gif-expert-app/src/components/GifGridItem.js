import React from 'react'
import PropTypes from "prop-types";

export default function GifGridItem({ title, url }) {
    return (
        <div className="card animate__animated animate__fadeInUp">
            <h2>{title}</h2>
            <img src={url} alt={title} />
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

GifGridItem.defaultProps = {
    title: 'Sooy el titulo',
    url: 'https://facebook.com'
}