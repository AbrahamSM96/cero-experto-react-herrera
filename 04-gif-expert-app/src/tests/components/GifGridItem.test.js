import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('test in GifGridItem component', () => {
    const title = 'soy titulo de prueba';
    const url = 'https://google.com '
    let wrapper = shallow(<GifGridItem title={title} url={url} />)
    test('should render the component correctly', () => {

        expect(wrapper).toMatchSnapshot()
    });
    test('should have a <h2> </h2> with the title', () => {
        const h2 = wrapper.find('h2');
        expect(h2.text().trim()).toBe(title)
    });

    test('should have an image equal to url', () => {
        const img = wrapper.find('img')
        // console.log(img.prop('src'), 'img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    });

    test('should animate__fadeInUp', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeInUp')).toBe(true)
    });
});