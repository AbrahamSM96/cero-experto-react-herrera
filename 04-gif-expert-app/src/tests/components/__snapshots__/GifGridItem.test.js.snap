// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGridItem component should render the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <h2>
    soy titulo de prueba
  </h2>
  <img
    alt="soy titulo de prueba"
    src="https://google.com "
  />
</div>
`;
